### PR TITLE
CDAP-14303 fix dataproc singlenode cluster provision

### PR DESCRIPTION
**Pull request by yaojiefeng**
JIRA: https://issues.cask.co/browse/CDAP-14303
Build: https://builds.cask.co/browse/CDAP-DUT6641

Fix the singlenode dataproc cluster provision.

### CODE REVIEW

File: cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java [hunk 1]
```diff
@@ -73,6 +73,12 @@
 public class DataprocClient implements AutoCloseable {
   // something like 2018-04-16T12:09:03.943-07:00
   private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss.SSSX");
+  // This property is needed to be able to provision a singlenode cluster on dataproc. Dataproc has an issue that
+  // it will treat 0 number of worker nodes as the default number, which means it will always provision a cluster with
+  // 2 worker nodes if this property is not set.
+  private static final Map<String, String> ADDITIONAL_PROPERTIES =
```
Comment (albertshau) [inline cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java]: you can just put this inline, there is no need for this additional map.

File: cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java [hunk 2]
```diff
@@ -73,6 +73,12 @@
 public class DataprocClient implements AutoCloseable {
   // something like 2018-04-16T12:09:03.943-07:00
   private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss.SSSX");
+  // This property is needed to be able to provision a singlenode cluster on dataproc. Dataproc has an issue that
+  // it will treat 0 number of worker nodes as the default number, which means it will always provision a cluster with
+  // 2 worker nodes if this property is not set.
```
Comment (karth295) [inline cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java]: Consider linking the docs as a reference: https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/single-node-clusters

File: cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java (hunk 1 above)
Comment (yaojiefeng) [reply cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java]: Moved

File: cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java (hunk 2 above)
Comment (yaojiefeng) [reply cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java]: Thanks, added this link to the comment

**Review by albertshau — DISMISSED** [review]
lgtm

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**